### PR TITLE
update README.mkd

### PR DESCRIPTION
**Pull request by mattn**
@haya14busa could you please take a look?

### CODE REVIEW

**Comment by haya14busa** [comment]
LGTM

File: README.mkd [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+
+This parser provide same feature for following languages.
+
+* Vim script
+* Python
+* Javascript 
+
+## Example
+
+All of interfaces are provided from vimlparser module. make StringReader to break a string into tokens, and VimLParser to parse into AST, Compiler to compile nodes.
```
Comment (haya14busa) [inline README.mkd]: actually, StringReader doesn't break a string into tokens. But it's just string reader for internal usage.

maybe just remove this paragraph?

File: README.mkd [hunk 2]
```diff
@@ -1,6 +1,39 @@
...
+# VimL Parsers
 
 [![Build Status](https://travis-ci.org/vim-jp/vim-vimlparser.svg?branch=master)](https://travis-ci.org/vim-jp/vim-vimlparser) [![codecov](https://codecov.io/gh/vim-jp/vim-vimlparser/branch/master/graph/badge.svg)](https://codecov.io/gh/vim-jp/vim-vimlparser)
 
+This is Vim script (a.k.a. VimL) parsers.
 
-TODO: write something.
+## Features
+
+* The string tokenizer allows scripts to break a string into tokens.
```
Comment (haya14busa) [inline README.mkd]: ditto

File: README.mkd (hunk 1 above)
Comment (haya14busa) [reply README.mkd]: e.g. Tokenizer for expression is ExprTokenizer.

**Comment by mattn** [comment]
Thanks for your review. Added example of ExprTokenizer.

File: README.mkd [hunk 3]
```diff
@@ -19,7 +19,9 @@ This parser provide same feature for following languages.
 
 ## Example
 
-All of interfaces are provided from vimlparser module. make StringReader to break a string into tokens, and VimLParser to parse into AST, Compiler to compile nodes.
+### Parse VimL into AST, and print S-expression
+
```
Comment (mattn) [inline README.mkd]: will remove this empty line

File: README.mkd [hunk 4]
```diff
@@ -1,6 +1,72 @@
...
+This above code output following.
+
+```
+(let = s:message (printf "hello %d" (+ 1 (* 2 3))))
+```
+
+### Parse VimL into AST, and print S-expression
+
+ExprTokenizer break a string buffer into tokens for expression.
```
Comment (haya14busa) [inline README.mkd]: If I understand correctly, ExprTokenizer is expected to be used for tokenize "expression".

This example seems to work but I think we should not write this example.
If we want to do this, we should rename the tokenizer and need to maintain "tokenizer" itself too.

I feel it's too early to make "tokenizer" as public API. Can we remove it from README?

File: README.mkd [hunk 5]
```diff
@@ -1,6 +1,72 @@
...
+echo join(c.compile(p.parse(r)), "\n")
+```
+
+This above code output following.
+
+```
+(let = s:message (printf "hello %d" (+ 1 (* 2 3))))
+```
+
+### Parse VimL into AST, and print S-expression
```
Comment (haya14busa) [inline README.mkd]: [nit] this header is wrong.

File: README.mkd [hunk 6]
```diff
@@ -6,7 +6,7 @@ This is Vim script (a.k.a. VimL) parsers.
 
 ## Features
 
-* The string tokenizer allows scripts to break a string into tokens.
+* The tokenizer allows scripts to break a string into tokens.
```
Comment (haya14busa) [inline README.mkd]: Remove this?

**Comment by mattn** [comment]
Thank you

**Pull request merged**